### PR TITLE
PWA

### PR DESCRIPTION
**Pull request by jbazant**
# Just some checks
- [x] No CSS file was changed or cache is busted.
- [x] No TS file was changed or cache is busted.
- [x] Styled sitemap works.
- [x] todo.md was updated accordingly.

### CODE REVIEW

File: src/_pwa-files.json.njk [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    {%- for page in collections.all %}
+      {%- if not page.data.excludeFromPWAPrecache and page.url %}
+    "{{ page.url | url }}",
+      {%- endif %}
+    {%- endfor %}
+    {%- for asset in pwa.assetsToPrecache %}
+    "{{ asset | url }}"{% if not loop.last %},{% endif %}
+    {%- endfor %}
+  ]
+}
```
Comment (c0mtru1se) [inline src/_pwa-files.json.njk:21]:
```suggestion
}

```

File: src/offline.njk [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+permalink: /offline.html
+eleventyExcludeFromCollections: true
+dynamicPermalink: false
+templateClass: offline-page
+title: Offline
+---
+<section class="offline-section">
+  <div class="offline-info">
+    <h1>Jste Offline</h1>
+    <p>To je sice trochu problém, ale tenhle web by měl trochu fungovat i&nbsp;tak. Co takhle zkusit to zatím třeba znovu
```
Comment (c0mtru1se) [inline src/offline.njk:12]: **nitpick (non-blocking)**: Are those `&nbsp;`s IDE created? I believe they are not needed here.

Comment (jbazant) [reply src/offline.njk:12]: I am not sure, why do you think this. Can you elaborate?

And no. Those are are written the old school way.

File: src/ts/utils/registerSW.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export function registerSW(): void {
+  if (location.hostname !== 'localhost' && 'serviceWorker' in navigator) {
+    if ('serviceWorker' in navigator) {
```
[the remark this answers is not in the document]
Comment (jbazant) [reply src/ts/utils/registerSW.ts]: Thanks! Fixed.

File: src/_pwa-files.json.njk (hunk 1 above)
Comment (jbazant) [reply src/_pwa-files.json.njk]: Fixed.

File: src/offline.njk (hunk 2 above)
Comment (c0mtru1se) [reply src/offline.njk:12]: Yeah, sorry, I've missed that it would be nice to keep `i tak` together.

**Pull request merged**